src/components/ProjectSettingsTabs/types.ts

```diff
@@ -1,17 +1,32 @@
+import { FormDataProps } from "components/SpruceForm";
 import { ProjectSettingsTabRoutes } from "constants/routes";
-import {
-  ProjectGeneralSettingsFragment,
-  RepoGeneralSettingsFragment,
-} from "gql/generated/types";
-import { GeneralFormState } from "./GeneralTab/formState";
+import { ProjectSettingsInput } from "gql/generated/types";
+import { PartialRecord } from "types/utils";
+import * as general from "./GeneralTab";
 
 export type FormStateMap = {
-  [ProjectSettingsTabRoutes.General]: GeneralFormState;
+  [ProjectSettingsTabRoutes.General]: general.FormState;
 };
 
-export type GeneralTabProps = {
-  projectData?: ProjectGeneralSettingsFragment;
-  projectId?: string;
-  repoData?: RepoGeneralSettingsFragment;
-  useRepoSettings: boolean;
+// TODO: Convert PartialRecord to Record once all tabs have been implemented.
```

Comment:
Should we create a ticket for this so that we don't forget, or nah?

Reply:
This is purely an internal nit and will work as designed if the change isn't made, so I think a ticket seems like overkill.

Reply:
that makes sense!